src/psyclone/psyir/symbols/datatypes.py

```diff
@@ -86,22 +90,34 @@ def __init__(self, declaration_txt):
                 format(type(declaration_txt).__name__))
         self._declaration = declaration_txt
 
+    @abc.abstractmethod
     def __str__(self):
-        '''
-        :returns: a description of this unknown type.
-        :rtype: str
-        '''
-        return "UnknownType('{0}')".format(self._declaration)
+        ''' Abstract method that must be implemented in subclass. '''
 
     @property
     def declaration(self):
         '''
-        :returns: the original declaration of the symbol.
+        :returns: the original declaration of the symbol. This is obviously \
+                  language specific and hence this class must be subclassed.
         :rtype: str
         '''
         return self._declaration
 
 
+class UnknownFortranType(UnknownType):
+    '''
+    Indicates that the Fortran type declaration is not supported by the PSyIR.
```

Comment:
Is is not just 'type' declarations here, is any declaration which can't be properly parsed, right? If so I would remove 'type' from the docstring.

There is (and was before this PR) the same issue in the UnknownType class docstring.

Reply:
Oh yes, thanks. I've fixed both.